nntrainer/src/tensor.cpp

```diff
@@ -775,4 +811,58 @@ Tensor Tensor::standardization() const {
 
   return result;
 }
+
+ExternalLoopInfo Tensor::computeExternalLoop(const Tensor &m) {
+  if (m.length() > this->length())
+    throw exception::not_supported("broadcasting *this is not supported");
+
+  const TensorDim m_dim = m.getDim();
+
+  ExternalLoopInfo e;
+
+  for (unsigned int i = 0; i < MAXDIM; ++i) {
+    if (m_dim.getTensorDim(i) == 1) {
+      e.strides[i] = 0;
```

Comment:
Add a contructor for `ExternalLoopInfo` inside that struct where these values are initialized (may be to 0).
Then remove this.

Reply:
This `if` statement checks if the given shape is broadcastable. I could set `0` as the default stride. but IMO, setting strides here shows the mechanism better in that if TensorDim is equal to 1, it sets stride to 0.

Reply:
`If TensorDim == 1, stride is used as default initialized value of 0`.
In c++, using struct/class, its best to have constructor initializing its members